sml/src/sml_time.c

```diff
@@ -51,8 +51,29 @@ sml_time *sml_time_parse(sml_buffer *buf) {
 	tme->tag = sml_u8_parse(buf);
 	if (sml_buf_has_errors(buf)) goto error;
 
-	tme->data.timestamp = sml_u32_parse(buf);
-	if (sml_buf_has_errors(buf)) goto error;
+	int type = sml_buf_get_next_type(buf);
+	switch(type) {
+	case SML_TYPE_UNSIGNED:
+		tme->data.timestamp = sml_u32_parse(buf);
+		if (sml_buf_has_errors(buf)) goto error;
+		break;
+	case SML_TYPE_LIST:
+	{
```

Comment:
warum Klammern in diesem fall und nicht im ersten case?
kannst Du einen Kommentar einfühgen für welchen Zähler wir das brauchen?

Reply:
In dem Code von @peterevertz war direkt nach dem `case SML_TYPE_LIST:` eine Variablendeklaration.  Dann braucht man `{`. Würde jetzt vmtl. ohne gehen.
Kommentar kann ich noch einfügen. Gute Idee.

Reply:
Die Klammern sind nötig damit die Variablen darin nicht die "Umgebung" verseuchen. Ist vielleicht nicht nötig, aber guter Stil.